textfsm.py

```diff
@@ -194,8 +210,8 @@ def OnAssignVar(self):
           match = None
       # If the List-value regex has match-groups defined, add the resulting dict to the list
       # Otherwise, add the string that was matched
-      if match and match.groupdict():
-          self._value.append(match.groupdict())
+      if match and match.capturesdict():
```

Comment:
capturesdict should only be called in the context of a Repeated match and List behavior is left unchanged. This can should probably be in the 'Repeated' class.